### DISCUSSION TITLE
Rustdesk shows workstation offline when its screen is off

### DISCUSSION THREAD

**Discussion by vieira-temes** [rustdesk/rustdesk#2159] [General]
**Describe the bug you encountered:**
Most of our workstations in the studio turn their screens off after 10 min. That's standard and recommended Windows behavior. At that point, Rustdesk can no longer connect to that workstation showing it offline.

...

**What did you expect to happen instead?**
Since the workstations is not offline, just its screen is off, Rustdesk shouldn't declare that workstation offline and just connect to it.
...

**How did you install `RustDesk`?**

Downloaded and executed this file: https://github.com/rustdesk/rustdesk/releases/download/1.1.9/rustdesk-1.1.9-x64.exe

---

**RustDesk version and environment**

1.1.9. Windows 10 fully updated, both on the remote and local computers.

**Comment by rustdesk** [comment]
the problem is what we can do?

**Reply by vieira-temes** [reply]
What do you mean? This is clearly a bug. The remote workstation is not suspended. The services aren't stopped. It's accesible for anyone on the network, including Rustdesk. Any attempt to do anything with that workstation will turn the screen on, from moving the mouse to touch the keyboard or a service doing something. The service running on the remote workstation should be answering as usual. Why is it not?

This does not affect any other remote access software I've ever used, be it VNC-type, RDP-type or Anydesk, Teamviewer, Google Chrome Remote... Anything!

**Reply by rustdesk** [reply]
help wanted

**Reply by rustdesk** [reply]
is it related to that we are using udp to keep alive?

**Reply by vieira-temes** [reply]
I have no idea.

**Reply by vieira-temes** [reply]
BTW: This is a proper issue. I don't understand why have you moved it to a "general conversation".

**Reply by rustdesk** [reply]
This move is not correct, I admit. Let me revert back to issue,  though I still have no clue.

**Comment by vieira-temes** [comment]
I'll try to make more tests to see if I can find a pattern that might help. Will also try to reproduce the issue on Linux and MacOS, both as initiator and target of the connection.

**Reply by rustdesk** [reply]
Great!

**Reply by rustdesk** [reply]
selfhost?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/2166

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/2166#issuecomment-1317998837
